autopts/utils.py

```diff
@@ -477,6 +490,70 @@ def log_memory_usage():
     logging.debug(f"Memory usage: {mem_usage:.2f} MB")
 
 
+def extract_wid_testcases_to_csv():
+
+    # Example log block format:
+    # BEGIN OnImplicitSend:
+    # wid: 35
+    # test_case_name: GAP/ADV/BV-01-C
+    # ...
+    # END OnImplicitSend
+
+    profile_wid_map = defaultdict(lambda: defaultdict(set))
+
+    for log_file in LOG_DIR.rglob("*.log"):
+        logging.debug(f"Processing log file: {log_file}")
+        try:
+            with log_file.open(encoding='utf-8', errors='ignore') as f:
+                lines = f.readlines()
+        except Exception as e:
+            logging.warning(f"Failed to read {log_file}: {e}")
+            continue
+
+        # Track wheter inside an OnImplicitSend block
+        in_implicit_send_block = False
+        wid = None
+        test_case_name = None
+
+        for line in lines:
+            stripped = line.strip()
+
+            # Detect start of a new block
+            if not in_implicit_send_block and stripped.startswith("BEGIN OnImplicitSend:"):
+                in_implicit_send_block = True
+                wid = None
+                test_case_name = None
+
+            elif in_implicit_send_block:
+                # Match and extract WID and TC fields
+                wid_match = WID_REGEX.match(stripped)
+                tc_name_match = TC_NAME_REGEX.match(stripped)
+
+                if wid_match:
+                    wid = wid_match.group(1)
+                elif tc_name_match:
+                    test_case_name = tc_name_match.group(1)
+                elif stripped == "END OnImplicitSend":
+                    # End of block - store data if complete
+                    if wid and test_case_name:
+                        profile = test_case_name.split('/')[0] if '/' in test_case_name else "UNKNOWN"
+                        profile_wid_map[profile][wid].add(test_case_name)
```

Comment:
Should we log an error if the profile is UNKNOWN?

Reply:
Added log